packages/rum-core/src/domain/contexts/pageStateContexts.ts

```diff
@@ -0,0 +1,157 @@
+import type { Duration, RelativeTime } from '@datadog/browser-core'
+import {
+  isExperimentalFeatureEnabled,
+  noop,
+  elapsed,
+  addDuration,
+  addEventListeners,
+  addEventListener,
+  relativeNow,
+  DOM_EVENT,
+} from '@datadog/browser-core'
+
+export const PAGE_STATE_TIMELINE_MAX_LENGTH = 2500
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Having to loop on 2500 states every time a resource is collected might impact performaces, maybe consider using a smaller limit? `100` could be enough, no?

Reply:
I'll try with 500